include/swift/SIL/SILInstruction.h

```diff
@@ -1222,12 +1222,14 @@ class StringLiteralInst final : public LiteralInst,
 private:
   unsigned Length;
   Encoding TheEncoding;
+  bool NullTerminated;
 
   StringLiteralInst(SILDebugLocation *DebugLoc, StringRef text,
-                    Encoding encoding, SILType ty);
+                    Encoding encoding, bool nullTerminate, SILType ty);
```

Comment:
Or, at least, allocate the null terminator into the StringLiteralInst's tail buffer, so we don't need to check this extra flag everywhere.